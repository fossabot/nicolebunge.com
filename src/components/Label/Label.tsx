/* eslint-disable jsx-a11y/label-has-associated-control */

import classNames from 'classnames';
import React, { HTMLAttributes } from 'react';
import * as styles from './Label.module.css';

export interface LabelProps extends HTMLAttributes<HTMLLabelElement> {
  htmlFor: string;
}

const Label: React.FC<LabelProps> = (props) => {
  const { className, ...otherProps } = props;

  return <label className={classNames(className, styles.label)} {...otherProps} />;
};

export default Label;
