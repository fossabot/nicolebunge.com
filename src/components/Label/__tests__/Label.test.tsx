/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Label from '..';

describe('Label component', () => {
  it('renders correctly', () => {
    const { container } = render(<Label htmlFor="name">Label</Label>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
