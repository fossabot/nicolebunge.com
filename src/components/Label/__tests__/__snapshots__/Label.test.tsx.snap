// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component renders correctly 1`] = `
<label
  class=""
  for="name"
>
  Label
</label>
`;
